paper.md

```diff
@@ -0,0 +1,144 @@
+has been a propensity to treat each build system as a one-off use case. This is
+understandable since build systems have tended to be relatively small (*i.e.*,
+less than ~1K lines of code) and tightly coupled to the identity of the package.
+With build system complexity at an all-time high, there is now an
+increasing desire (at least within scientific computing communities) to treat
+the underlying build system infrastructure as code. This means that the
+build system should be modularized, and those modules should be documented,
+tested, and reusable. With the popularity of C/C++ for high-performance
+computing, "build system" is increasingly becoming synonymous with CMake
+(*N.B.* CMake is often described as a build system generator; however, since
+the typical CMake workflow encapsulates running the build system, the
+present manuscript ignores the "generator" distinction). Therefore with the
+advent of the exascale era, there is a desperate need for a robust CMake
+development ecosystem.
+
+CMake already contains a number of tools and features that facilitate
+development of the target software package. For example, CMake's
+`find_package` module simplifies dependency management, and the `CTest` package
+eases the process of testing the resulting software. Additional CMake tools
+can be created by writing CMake modules. While the CMake language is
+flexible and relatively simple, it is not without its pitfalls. Unfortunately,
+tools  to facilitate the development of CMake modules are relatively sparse.
+Here we introduce CMinx, a tool for generating API documentation for CMake
+modules. API documentation is arguably one of the most basic elements of a
+software development ecosystem, and it is our hope that CMinx will serve as the
+foundation for a robust CMake development ecosystem.
+
+Anecdotal evidence [@official_solution] indicates that the developers of the
+CMake language (*i.e.*, Kitware) internally write their API documentation
+using reST and Sphinx. Following best practices, this reST documentation
+resides next to the described CMake code. Kitware has also written a
+Sphinx plugin that makes it easy to extract the API documentation as part of a
+normal Sphinx workflow. This Sphinx plugin is distributed with the source code
+for the CMake interpreter and is also available in a GitHub repository mirror
+[@sphinx_plugin]. For completeness, we note that similar Sphinx plugins
+[@official_sphinx_domain; @marco_koch] have been independently developed but
+appear to now be abandoned.
+
+To our knowledge, all of the aforementioned Sphinx plugins simply extract the
+reST API documentation verbatim. Notably, this means the developer is
+responsible
+for manually: adding the function/macro signatures to the documentation, listing
+the function's parameters, and the overall formatting (*e.g.*, titles and
+subtitles, creation of parameter tables, underline/overline consistency). For
```

Comment:
Again, the examples are specific in my perspective. Also, the colon appears a bit stray to me.
```suggestion
for manually adding the function/macro signatures to the documentation, listing
the function's parameters, and the overall formatting. For
```